### PR TITLE
Add x86_64 dependency note to pip installation

### PR DESCRIPTION
**Pull request by treyra**
Currently non x86_64 linux architectures are not supported, see #7097 for request to change this. This can lead to installation confusion, as jax will install, but jaxlib will not. For example see #12307. 

If this can be more clearly phrased or explained, let me know.

### CODE REVIEW

**Comment by jakevdp** [comment]
Thanks! Should we add the same caveat under the GPU instructions?

**Comment by mjsML** [comment]
For the current state I think this makes sense ... It would change soon-ish with Grace and Grace-Hopper

**Comment by treyra** [comment]
@jakevdp good point, I added that too.

**Comment by mjsML** [comment]
Forgot about jetsons too 😅

**Comment by treyra** [comment]
Yeah, a Jetson Orin was what I was trying to install Jax on and had to build from source for. The good news is building from source was pretty easy! But this would have saved me some time debugging why pip appeared to work, but jaxlib didn't install.

**Review by jakevdp — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -413,7 +413,7 @@ pip install --upgrade "jax[cpu]"
 ```
 
 On Linux, it is often necessary to first update `pip` to a version that supports
-`manylinux2014` wheels.
+`manylinux2014` wheels. Note these existing wheels are currently for `x86_64` architectures only. Trying to pip install with other architectures may lead to `jaxlib` not being installed alongside `jax`, although `jax` may successfully install.
```
Comment (hawkinsp) [inline README.md:416]: The wording here is a bit misleading, because we do build and release Mac ARM wheels. So you should probably put "Linux" somewhere. Perhaps reword this change as:
"Note that on Linux, we release wheels for the `x86_64` architecture only at the moment. Other architectures require building from source."

Comment (treyra) [reply README.md]: Good point, I reworded it to emphasize that caveat applies to Linux architectures only.

**Review by hawkinsp — APPROVED** [review]

**Comment by treyra** [comment]
Let me know if there is anything else needed from me before this is merged.

I don't seem to have permissions to re-run the Read the Docs build, but from reading the output, it seems to be just two warnings, neither related to my edits. The first was that https://docs.scipy.org/doc/scipy-1.8.1/objects.inv wasn't fetchable, which seems to have been temporary and resolved. The other was undefined label: ndimage-interpolation-modes in jax._src.scipy.ndimage.map_coordinates. If wanted I can make an empty commit to force it to re-run.

**Comment by jakevdp** [comment]
Thanks @treyra - one last thing, could you please squash the changes into a single commit? (See https://jax.readthedocs.io/en/latest/contributing.html#single-change-commits-and-pull-requests) Thanks!

**Comment by treyra** [comment]
Done! Thanks @jakevdp for walking me through how to contribute for the first time. Hopefully in the future I can make some more substantial contributions.

**Comment by jakevdp** [comment]
Thanks for adding this clarification! Sorry it took so long to get this merged 😀
